### PR TITLE
Services-contracts-schemas

### PR DESCRIPTION
**Pull request by tomasruizr**

### CODE REVIEW

**Review by Rotorsoft — COMMENTED** [review]
LGTM, just a comment about optimizing the services filter before querying

File: libs/eventually-broker/src/routes/contracts.ts [hunk 1]
```diff
@@ -1,14 +1,52 @@
...
+      return result;
+    }, {} as Record<string, ContractsViewModel>)
+  }
+
+  return servicesContracts;
+}
+
+router.get("/all", async (req, res) => {
+  const servicesDefinitions = await subscriptions().loadServices();
+  const servicesContracts = await getServiceContracts(servicesDefinitions);
```
[the remark this answers is not in the document]
Comment (tomasruizr) [reply libs/eventually-broker/src/routes/contracts.ts]: Totally Agree!!.. fixed!

File: libs/eventually-broker/src/routes/contracts.ts [hunk 2]
```diff
@@ -1,14 +1,53 @@
...
+  }
+  
+  let servicesDefinitions = (await subscriptions().loadServices())
+  
+  servicesDefinitions = !services ? servicesDefinitions : servicesDefinitions
+    .filter((service) => !services.includes(service.id));
+  
+    const servicesContracts = await getServiceContracts(servicesDefinitions);
+
+  res.send(filterNames(servicesContracts, {
```
Comment (Rotorsoft) [inline libs/eventually-broker/src/routes/contracts.ts:40]: pass names only to filterNames?

**Review by Rotorsoft — COMMENTED** [review]
one more

**Review by Rotorsoft — APPROVED** [review]
LGTM

**Pull request merged**